libbeat/common/transport/tlscommon/validhostname.go

```diff
@@ -47,7 +47,69 @@
 
 package tlscommon
 
-import "strings"
+import (
+	"strings"
+	"unicode/utf8"
+)
+
+func matchHostnames(pattern, host string) bool {
+	pattern = toLowerCaseASCII(pattern)
+	host = toLowerCaseASCII(strings.TrimSuffix(host, "."))
+
+	if len(pattern) == 0 || len(host) == 0 {
+		return false
+	}
+
+	patternParts := strings.Split(pattern, ".")
+	hostParts := strings.Split(host, ".")
+
+	if len(patternParts) != len(hostParts) {
+		return false
+	}
+
+	for i, patternPart := range patternParts {
+		if i == 0 && patternPart == "*" {
+			continue
+		}
+		if patternPart != hostParts[i] {
+			return false
+		}
+	}
+
+	return true
+}
+
+// toLowerCaseASCII returns a lower-case version of in. See RFC 6125 6.4.1. We use
+// an explicitly ASCII function to avoid any sharp corners resulting from
+// performing Unicode operations on DNS labels.
+func toLowerCaseASCII(in string) string {
```

Comment:
It seems fairly easy to add unit tests for for these two methods. Could we add them too?

Reply:
I am adding tests ATM, but not for these specific function, but for the whole VerifyConnection callback.

Reply:
Great, thanks! If possible, I'd add unit tests for the calculation and verification of the lowerCaseASCII hostname, just in case we missed any edge case for the conditional branches inside both methods.